src/update/UpdateManager.cpp

```diff
@@ -130,6 +130,42 @@ void UpdateManager::GetUpdates(bool ignoreNetworkErrors)
 
                 if (code == wxID_YES)
                 {
+                    const audacity::network_manager::Request downloadRequest(mVersionPatch.download.ToStdString());
+                    auto downloadResponse = audacity::network_manager::NetworkManager::GetInstance().doGet(downloadRequest);
+
+                    downloadResponse->setRequestFinishedCallback([downloadResponse, this](audacity::network_manager::IResponse*) {
+                        mProgressDialog.reset();
+                        }
+                    );
+
+                    downloadResponse->setDownloadProgressCallback(
+                        [downloadResponse, this](int64_t current, int64_t expected) {
+
+                            static std::once_flag flag;
+                            std::call_once(flag, [this] {
+                                const auto title = XO("Audacity update")
+                                    .Translation();
+
+                                auto audacityPatchFilename = wxFileName(mVersionPatch.download).GetName();
+                                const auto message = XO("Downloading %s")
+                                    .Format(audacityPatchFilename)
+                                    .Translation();
+
+                                mProgressDialog.reset(new wxGenericProgressDialog(title, message,
```

Comment:
You might use the recently added BasicUI::MakeProgress() to make this bit of code "toolkit-neutral."  But it is not important, yet, to do that here.

Reply:
MakeProgress or MakeGenericProgress, as appropriate -- I may have overspecified it.

If there is no "denominator" for the "numerator" -- no real estimate of the incomplete work, then the Generic progress is the right one.

Ah, but the download callback is supplied "current" and "expected."  So that means, don't use Generic?